BWT/BWT/Program.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            if (stringToBWT[i % stringToBWT.Length] > stringToBWT[j % stringToBWT.Length])
+            {
+                return WichStringIsBigger.first;
+            } 
+            else if (stringToBWT[i % stringToBWT.Length] < stringToBWT[j % stringToBWT.Length])
+            {
+                return WichStringIsBigger.second;
+            }
+
+            ++i;
+            ++j;
+            ++k;
+        }
+        return WichStringIsBigger.same;
+    }
+
+    //  Sorting by inserts
+    public static void InsertSort(string stringToBWT, int[] arrayPositions, int startArray, int endArray)
+    {
+        for (int i = startArray + 1; i <= endArray; ++i)
+        {
+            int j = i;
+            while (j >= startArray + 1 && ComprassionStrings(stringToBWT, (arrayPositions[j - 1] + 1) % arrayPositions.Length, (arrayPositions[j] + 1) % arrayPositions.Length) == WichStringIsBigger.first)
+            {
+                (arrayPositions[j - 1], arrayPositions[j]) = (arrayPositions[j], arrayPositions[j - 1]);
+
+                --j;
+            }
+        }
+    }
+
+    // Finding a reference element
+    public static int Partition(string stringToBWT, int[] arrayPositions, int startArray, int endArray)
+    {
+        int pivot = arrayPositions[endArray];
+        int i = startArray;
+
+        for (int j = startArray; j < endArray; ++j)
+        {
+            WichStringIsBigger result = ComprassionStrings(stringToBWT, (arrayPositions[j] + 1) % arrayPositions.Length, (pivot + 1) % arrayPositions.Length);
+            if (result == WichStringIsBigger.second || result == WichStringIsBigger.same)
+            {
+                (arrayPositions[j], arrayPositions[i]) = (arrayPositions[i], arrayPositions[j]);
+
+                ++i;
+            }
+        }
+        (arrayPositions[i], arrayPositions[endArray]) = (arrayPositions[endArray], arrayPositions[i]);
+        return i;
+    }
+
+    // Quick sort, which sorts an array of indexes by comparing rows obtained by cyclic permutations
+    public static void QSort(string stringToBWT, int[] arrayPositions, int startArray, int endArray)
+    {
+        if (endArray - startArray + 1 <= 10)
+        {
+            InsertSort(stringToBWT, arrayPositions,startArray, endArray);
+            return;
+        }
+        if (startArray < endArray)
+        {
+            int positionElement = Partition(stringToBWT, arrayPositions, startArray, endArray);
+            QSort(stringToBWT, arrayPositions, startArray, positionElement - 1);
+            QSort(stringToBWT, arrayPositions, positionElement + 1, endArray);
+        }
+    }
+
+    // Shell for running quick sort
+    public static void QuickSort(string stringToBWT, int[] arrayPositions)
+    {
+        QSort(stringToBWT, arrayPositions, 0, arrayPositions.Length - 1);
+    }
+
+    // String compression by the Burrows-Wheeler algorithm
+    public static string CompressionBWT(string stringToBWT, ref int firstPosition)
```

Comment:
Можно возвращать сразу пару,
```suggestion
    public static (string result, int firstPosition) CompressionBWT(string stringToBWT)
```